base/show.jl

```diff
@@ -322,6 +327,11 @@ function isidentifier(s::AbstractString)
     end
     return true
 end
+"""
+    isidentifier(s) -> Bool
+
+Returns whether symbol is vaild identifier.
```

Comment:
Your formulation was much better than mine. I just answered in a hurry, to note that there was two methods. 

You wouldn't use `isidentifier()` to parse, but it is useful when you want to test if you can print a symbol using the `:sym` syntax, or if you need `@eval(symbol("sym") ) ` or (`:"sym"`, if that is implemented) . @armgong also want to use it to be able to do some name mangling in a ffi (AFAIU).